src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig

```diff
@@ -11,19 +11,21 @@
                 </tr>
             </thead>
             <tbody>
-            {% for channelPricing in product.variants.first.channelPricings %}
-                <tr>
-                    <td class="five wide gray text">
-                        <strong>{{ channelPricing.channelCode|sylius_channel_name }} </strong>
-                    </td>
-                    <td>{{ money.format(channelPricing.price, product.channels.first.baseCurrency) }}</td>
-                    {% if channelPricing.originalPrice != null %}
-                        <td>{{ money.format(channelPricing.originalPrice, product.channels.first.baseCurrency) }}</td>
-                    {% else %}
-                        <td>-</td>
-                    {% endif %}
-                </tr>
-            {% endfor %}
+                {% set currencies = sylius_channels_currencies() %}
+                {% for channelPricing in product.variants.first.channelPricings %}
+                    {% set channel = channelPricing.channelCode|sylius_channel %}
```

Comment:
This filter will always call the database for every channel, there is configured for a given variant.

I'd like to have a custom controller action which renders this template given a variant id and optimises these calls - this will also remove the need for the custom filter, which feels a bit hacky. We can achieve it by using `render(controller())` or `render(url())` instead of `include` in Twig (for this template).